internal/batches/service/service.go

```diff
@@ -380,9 +381,53 @@ func (svc *Service) ParseBatchSpec(data []byte) (*batcheslib.BatchSpec, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "parsing batch spec")
 	}
+	if err = validateMount(dir, spec, isRemote); err != nil {
+		return nil, errors.Wrap(err, "handling mount")
+	}
 	return spec, nil
 }
 
+const invalidMountCharacters = ","
+
+func validateMount(batchSpecDir string, spec *batcheslib.BatchSpec, isRemote bool) error {
+	for i, step := range spec.Steps {
+		for j, mount := range step.Mount {
+			if isRemote {
+				return errors.New("mounts are not support for server-side processing")
+			}
+			if strings.Contains(mount.Path, invalidMountCharacters) {
+				return errors.Newf("step %d mount path contains invalid characters", i+1)
+			}
+			if strings.Contains(mount.Mountpoint, invalidMountCharacters) {
+				return errors.Newf("step %d mount mountpoint contains invalid characters", i+1)
+			}
+			p := mount.Path
+			if !filepath.IsAbs(p) {
+				// Try to build the absolute path since Docker will only mount absolute paths
+				p = filepath.Join(batchSpecDir, p)
+			}
+			pathInfo, err := os.Stat(p)
+			if os.IsNotExist(err) {
+				return errors.Newf("step %d mount path %s does not exist", i+1, p)
+			} else if err != nil {
+				return errors.Wrapf(err, "step %d mount path validation", i+1)
+			}
+			if !strings.HasPrefix(p, batchSpecDir) {
+				return errors.Newf("step %d mount path is not in the same directory or subdirectory as the batch spec", i+1)
+			}
```

Comment:
Can we move the static checks in here into the lib parse method, so SSBC can benefit from the same validation? Only works for the ones that don't need to access the FS, of course.

Reply:
I'll need to wait for the `sourcegraph` PR to be merged so I can update `go.mod` to the latest version